pkg/apis/core/types.go

```diff
@@ -2235,9 +2229,8 @@ type Lifecycle struct {
 	// PreStop is called immediately before a container is terminated due to an
 	// API request or management event such as liveness/startup probe failure,
 	// preemption, resource contention, etc. The handler is not called if the
-	// container crashes or exits. The reason for termination is passed to the
```

Comment:
> The reason for termination is passed to the handler.

I don't know what this is referring to. I didn't find evidence that this is true or was true at some point

Reply:
Fixes #99250.